plugin/pkg/admission/runtimeclass/admission_test.go

```diff
@@ -56,6 +56,8 @@ func newOverheadValidPod(name string, numContainers int, resources core.Resource
 			core.ResourceName(core.ResourceCPU):    resource.MustParse("100m"),
 			core.ResourceName(core.ResourceMemory): resource.MustParse("1"),
 		}
+	} else if setEmptyOverHead {
+		pod.Spec.Overhead = core.ResourceList{}
```

Comment:
can we turn this into a function? the vast majority of time the `setEmptyOverhead` is bool, so I'd rather do
```
func setEmptyOverhead(pod *core.Pod) *core.Pod {
    pod.Spec.Overhead = core.ResourceList{}
    return pod
}
```

use it like

```
pod: setEmptyOverhead(newOverheadValidPod("no-resource-req-no-overhead", 1, getGuaranteedRequirements(), false)),
```

this should enable us to minimize the changes in this PR